net/Socket.hpp

```diff
@@ -362,26 +410,47 @@ class Socket
     /// Construct based on an existing socket fd.
     /// Used by accept() only.
     Socket(const int fd, Type type)
-        : _fd(fd)
+        : _type(type),
+          _clientPort(0),
+          _fd(fd),
+          _open(_fd >= 0),
+          _creationTime(std::chrono::steady_clock::now()),
+          _lastSeenTime(_creationTime),
+          _bytesSent(0),
+          _bytesRcvd(0)
     {
-        init(type);
+        init();
     }
 
     inline void logPrefix(std::ostream& os) const { os << '#' << _fd << ": "; }
 
     /// avoid doing a shutdown before close
     void setNoShutdown() { _noShutdown = true; }
 
+    /// Explicitly marks this socket closed, i.e. rejected from polling and potentially shutdown
+    void setClosed() noexcept { _open = false; }
+
+    /// Explicitly marks this socket and the given SocketDisposition closed
+    void setClosed(SocketDisposition &disposition) noexcept { setClosed(); disposition.setClosed(); }
+
 private:
-    void init(Type type)
+    /// Create socket of the given type.
+    /// \return >= 0 for a successfully created socket, -1 on error
+    static int createSocket(Type type);
+
+    std::string toStringImpl() const noexcept;
+
+    void init()
     {
-        if (type != Type::Unix)
+        if (_type != Type::Unix)
+        {
```

Comment:
Again changing this in lots of places is not my favourite change =)

Reply:
type became a private field (underscore) .. and I wanted to clarify access (see above).
The overall work was to clean up socket, side effects etc .. while reading and understanding the code.
Mixed field access is very often a PIA .. and nowadays our C++ "constexpr" getter or queries are validated and inlined, see e.g. isIPType(). Perhaps a matter of taster.